src/hacbs/components/ApplicationDetails/tabs/overview/visualization/hooks/useAppWorkflowData.ts

```diff
@@ -1,129 +1,582 @@
+          );
+        })
+      : [
+          emptyPipelineNode(
+            'no-static-environments',
+            'Static environments',
+            WorkflowNodeType.STATIC_ENVIRONMENT,
+            applicationIntegrationTestTasks,
+          ),
+        ];
+    const maxWidth = nodes.reduce((max, node) => Math.max(max, node.width), 0);
+    nodes.forEach((n) => (n.width = maxWidth));
+    return nodes;
+  }, [applicationIntegrationTestTasks, healthStatus, staticEnvironments]);
+
+  const staticEnvironmentGroup = React.useMemo(
+    () =>
+      allResourcesLoaded && testsGroup?.id
+        ? groupToPipelineNode(
+            'static-environments',
+            'Static environments',
+            WorkflowNodeType.STATIC_ENVIRONMENT,
+            expanded ? applicationIntegrationTestTasks : [testsGroup.id],
+            expanded,
+            expanded ? staticEnvironmentNodes?.map((c) => c.id) : undefined,
+            staticEnvironmentNodes,
+            staticEnvironments,
+            healthStatus,
+          )
+        : undefined,
+    [
+      allResourcesLoaded,
+      applicationIntegrationTestTasks,
+      expanded,
+      healthStatus,
+      staticEnvironmentNodes,
+      staticEnvironments,
+      testsGroup?.id,
+    ],
+  );
+
+  const lastStaticEnv = getLastEnvironments(staticEnvironments);
+
+  const releasePlanNodes = React.useMemo(() => {
+    const nodes = releasePlans.length
+      ? releasePlans.map((releasePlan) =>
+          resourceToPipelineNode(
+            releasePlan,
+            WorkflowNodeType.RELEASE,
+            lastStaticEnv,
+            pipelineRunStatus(releasePlan),
+          ),
+        )
+      : [
+          emptyPipelineNode(
+            'no-release-plans',
+            'Release plans',
+            WorkflowNodeType.RELEASE,
+            lastStaticEnv,
+          ),
+        ];
+    const maxWidth = nodes.reduce((max, node) => Math.max(max, node.width), 0);
+    nodes.forEach((n) => (n.width = maxWidth));
+    return nodes;
+  }, [lastStaticEnv, releasePlans]);
+
+  const releasePlanGroup = React.useMemo(
+    () =>
+      allResourcesLoaded
+        ? groupToPipelineNode(
+            'release-plans',
+            'Releases',
+            WorkflowNodeType.RELEASE,
+            expanded ? lastStaticEnv : [staticEnvironmentGroup?.id ?? ''],
+            expanded,
+            expanded ? releasePlanNodes?.map((c) => c.id) : undefined,
+            releasePlanNodes,
+            releasePlans,
+          )
+        : undefined,
+    [
+      allResourcesLoaded,
+      expanded,
+      lastStaticEnv,
+      staticEnvironmentGroup?.id,
+      releasePlanNodes,
+      releasePlans,
+    ],
+  );
+
+  const releasePlanTasks = React.useMemo(
+    () =>
+      expanded && releasePlanNodes?.length
+        ? releasePlanNodes.map((c) => c.id)
+        : [releasePlanGroup?.id ?? ''],
+    [releasePlanGroup?.id, releasePlanNodes, expanded],
+  );
+
+  const managedEnvironments = environments.filter(
```

Comment:
use `ReleasePlan` cr to for the managed environment. Related to my previous comment.